header.go

```diff
@@ -1859,6 +1872,9 @@ func (h *ResponseHeader) parseFirstLine(buf []byte) (int, error) {
 		}
 		return 0, fmt.Errorf("unexpected char at the end of status code. Response %q", buf)
 	}
+	if len(b) > n+1 && !bytes.Equal(b[n+1:], statusLine(h.statusCode)) {
+		h.SetStatusLine(b[n+1:])
+	}
```

Comment:
If you're going to parse it here then I suggest adding a getter method as well `func (h *ResponseHeader) StatusLine() []byte {`.